### PR TITLE
Interpolation Search 

### PR DESCRIPTION
**Pull request by hariprasadraja**
Added Interpolation Search

### CODE REVIEW

**Review by iamrajiv — DISMISSED** [review]

File: DIRECTORY.md [hunk 1]
```diff
@@ -147,6 +147,7 @@
 
 ## Searches
   * [Binary Search](https://github.com/TheAlgorithms/Go/blob/master/searches/binary_search.go)
+  * [Interpolationsearch](https://github.com/TheAlgorithms/Go/blob/master/searches/interpolationsearch.go)
```
Comment (siriak) [inline DIRECTORY.md:150]:
```suggestion
  * [Interpolation Search](https://github.com/TheAlgorithms/Go/blob/master/searches/interpolationsearch.go)
```

**Review by siriak — APPROVED** [review]
LGTM, thanks!

**Comment by siriak** [comment]
Let's wait for #320 to be merged first

**Comment by hariprasadraja** [comment]
Hi @raklaptudirm, @iamrajiv,  Resolved conflicts with the master branch by rebasing and updated my changes.

**Comment by raklaptudirm** [comment]
@siriak this pr needs rebasing.

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**